lib/puppet/provider/exec/powershell.rb

```diff
@@ -47,7 +47,7 @@ def self.upgrade_message
 
   def ps_manager
     debug_output = Puppet::Util::Log.level == :debug
-    Pwsh::Manager.instance(Pwsh::Manager.powershell_path, Pwsh::Manager.powershell_args, debug: debug_output)
+    Pwsh::Manager.instance(Pwsh::Manager.powershell_path, Pwsh::Manager.powershell_args, debug: debug_output, pipe_timeout: 300)
```

Comment:
what was the timeout previously?

Reply:
It defaults to 30 here: https://github.com/puppetlabs/ruby-pwsh/blob/main/lib/pwsh.rb#L37